In [1]:
import pandas as pd

In [2]:
menu_file = './data/menu_data.csv'
sales_file = './data/sales_data.csv'

In [4]:
menu_df = pd.read_csv(menu_file)
sales_df = pd.read_csv(sales_file)

In [6]:
menu_df.head()

item    category  \
0                       edamame  appetizers   
1                   house salad  appetizers   
2       cucumber sunomono salad  appetizers   
3  hiyashi wakame seaweed salad  appetizers   
4                 agedashi tofu  appetizers   

                                         description  price  cost  
0                   boiled soybeans, maldon sea salt    4.0     1  
1  mixed greens, cherry tomatoes, cucumber, house...    4.0     2  
2  cucumber, ponzu dressing, pickled ginger sprou...    5.0     2  
3  sweet soy-marinated wakame seaweed threads, pi...    5.0     2  
4  tempura battered tofu, katsuobushi, tsuyu brot...    5.0     2

In [7]:
sales_df.head()

Line_Item_ID        Date  Credit_Card_Number  Quantity  \
0             1  01-01-2017    7437926611570799         1   
1             2  01-01-2017    7437926611570799         1   
2             3  01-01-2017    8421920068932819         3   
3             4  01-01-2017    8421920068932819         3   
4             5  01-01-2017    4787310681569648         1   

              Menu_Item  
0      spicy miso ramen  
1      spicy miso ramen  
2     tori paitan ramen  
3     tori paitan ramen  
4  truffle butter ramen

In [8]:
sales_df.shape

(74124, 5)

In [9]:
menu_df.shape

(32, 5)

## The Count for each item

In [16]:
item_count = sales_df.groupby('Menu_Item').Quantity.sum()
type(item_count)

pandas.core.series.Series

## Get the Price for each of these items


In [ ]:
#pd.merge(sales_df,menu_df,on='id')
#pd.merge(left, right, how='inner', on=None, left_on=None, right_on=None, left_index=False, right_index=False, sort=True)

In [18]:
mnu_sales_df = pd.merge(menu_df, sales_df, how='inner', left_on='item', right_on='Menu_Item')

In [25]:
mnu_sales_df.head()

item category                                        description  \
0  nagomi shoyu   entree  shoyu seasoned clear double stock (organic chi...   
1  nagomi shoyu   entree  shoyu seasoned clear double stock (organic chi...   
2  nagomi shoyu   entree  shoyu seasoned clear double stock (organic chi...   
3  nagomi shoyu   entree  shoyu seasoned clear double stock (organic chi...   
4  nagomi shoyu   entree  shoyu seasoned clear double stock (organic chi...   

   price  cost  Line_Item_ID        Date  Credit_Card_Number  Quantity  \
0   11.0     5            39  01-01-2017    5567347994803336         1   
1   11.0     5            40  01-01-2017    5567347994803336         1   
2   11.0     5            51  01-01-2017    9720247648273415         3   
3   11.0     5            52  01-01-2017    9720247648273415         3   
4   11.0     5            57  01-01-2017    2925610854460099         2   

      Menu_Item  revenue  cogs  profit  
0  nagomi shoyu     11.0     5     6.0  
1  nagomi shoyu     11.0     5     6.0  
2  nagomi shoyu     33.0    15    18.0  
3  nagomi shoyu     33.0    15    18.0  
4  nagomi shoyu     22.0    10    12.0

In [26]:
mnu_sales_df.shape

(74124, 13)

In [21]:
mnu_sales_df['revenue'] = mnu_sales_df.price * mnu_sales_df.Quantity

In [24]:
mnu_sales_df['cogs'] = mnu_sales_df.cost * mnu_sales_df.Quantity
mnu_sales_df['profit'] = mnu_sales_df.revenue - mnu_sales_df.cogs

In [28]:
results = mnu_sales_df.groupby('item').agg({'Quantity': ['sum'], 'revenue': ['sum'], 'cogs':['sum'], 'profit':['sum']})

In [29]:
results

Quantity   revenue   cogs  \
                                                    sum       sum    sum   
item                                                                       
burnt garlic tonkotsu ramen                        9070  126980.0  54420   
miso crab ramen                                    8890  106680.0  53340   
nagomi shoyu                                       9132  100452.0  45660   
shio ramen                                         9180  100980.0  45900   
soft-shell miso crab ramen                         9130  127820.0  63910   
spicy miso ramen                                   9238  110856.0  46190   
tonkotsu ramen                                     9288  120744.0  55728   
tori paitan ramen                                  9156  119028.0  54936   
truffle butter ramen                               8982  125748.0  62874   
vegetarian curry + king trumpet mushroom ramen     8824  114712.0  61768   
vegetarian spicy miso                              9216  110592.0  46080   

                                                 profit  
                                                    sum  
item                                                     
burnt garlic tonkotsu ramen                     72560.0  
miso crab ramen                                 53340.0  
nagomi shoyu                                    54792.0  
shio ramen                                      55080.0  
soft-shell miso crab ramen                      63910.0  
spicy miso ramen                                64666.0  
tonkotsu ramen                                  65016.0  
tori paitan ramen                               64092.0  
truffle butter ramen                            62874.0  
vegetarian curry + king trumpet mushroom ramen  52944.0  
vegetarian spicy miso                           64512.0

spicy miso ramen {'01-count': 9238, '02-revenue': 110856.0, '03-cogs': 46190.0, '04-profit': 64666.0}
tori paitan ramen {'01-count': 9156, '02-revenue': 119028.0, '03-cogs': 54936.0, '04-profit': 64092.0}
truffle butter ramen {'01-count': 8982, '02-revenue': 125748.0, '03-cogs': 62874.0, '04-profit': 62874.0}
tonkotsu ramen {'01-count': 9288, '02-revenue': 120744.0, '03-cogs': 55728.0, '04-profit': 65016.0}
vegetarian spicy miso {'01-count': 9216, '02-revenue': 110592.0, '03-cogs': 46080.0, '04-profit': 64512.0}
shio ramen {'01-count': 9180, '02-revenue': 100980.0, '03-cogs': 45900.0, '04-profit': 55080.0}
miso crab ramen {'01-count': 8890, '02-revenue': 106680.0, '03-cogs': 53340.0, '04-profit': 53340.0}
nagomi shoyu {'01-count': 9132, '02-revenue': 100452.0, '03-cogs': 45660.0, '04-profit': 54792.0}
soft-shell miso crab ramen {'01-count': 9130, '02-revenue': 127820.0, '03-cogs': 63910.0, '04-profit': 63910.0}
burnt garlic tonkotsu ramen {'01-count': 9070, '02-revenue': 126980.0, '03-cogs': 54420.0, '04-profit': 72560.0}
vegetarian curry + king trumpet mushroom ramen {'01-count': 8824, '02-revenue': 114712.0, '03-cogs': 61768.0, '04-profit': 52944.0}
